File: backend/src/main/java/com/back/ourlog/global/security/filter/JwtAuthenticationFilter.java
package com.back.ourlog.global.security.filter;

import com.back.ourlog.global.security.jwt.JwtAuthenticationProvider;
import com.back.ourlog.global.security.jwt.JwtAuthenticationToken;
import jakarta.servlet.FilterChain;
import jakarta.servlet.ServletException;
import jakarta.servlet.http.HttpServletRequest;
import jakarta.servlet.http.HttpServletResponse;
import lombok.RequiredArgsConstructor;
import org.springframework.security.core.Authentication;
import org.springframework.security.core.context.SecurityContextHolder;
import org.springframework.stereotype.Component;
import org.springframework.web.filter.OncePerRequestFilter;

import java.io.IOException;

@Component
@RequiredArgsConstructor
public class JwtAuthenticationFilter extends OncePerRequestFilter {

    private final JwtAuthenticationProvider authenticationProvider;

    @Override
    protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response, FilterChain filterChain)
            throws ServletException, IOException {

        String header = request.getHeader("Authorization");

        if (header != null && header.startsWith("Bearer ")) {
            String token = header.substring(7);
            JwtAuthenticationToken authToken = new JwtAuthenticationToken(token);

            // 예외 발생 시 자동으로 AuthenticationEntryPoint로 감
            Authentication authResult = authenticationProvider.authenticate(authToken);
            SecurityContextHolder.getContext().setAuthentication(authResult);
        }

        filterChain.doFilter(request, response);
    }
}

File: backend/src/main/java/com/back/ourlog/global/security/filter/JwtAuthenticationFilter.kt
package com.back.ourlog.global.security.filter

import com.back.ourlog.global.security.jwt.JwtAuthenticationProvider
import com.back.ourlog.global.security.jwt.JwtAuthenticationToken
import jakarta.servlet.FilterChain
import jakarta.servlet.http.HttpServletRequest
import jakarta.servlet.http.HttpServletResponse
import org.springframework.security.core.context.SecurityContextHolder
import org.springframework.web.filter.OncePerRequestFilter

class JwtAuthenticationFilter (
    private val authenticationProvider: JwtAuthenticationProvider
) : OncePerRequestFilter() {

    override fun doFilterInternal(
        request: HttpServletRequest,
        response: HttpServletResponse,
        filterChain: FilterChain
    ) {
        val header = request.getHeader("Authorization")

        if (header != null && header.startsWith("Bearer ")) {
            val token = header.substring(7)
            val authToken = JwtAuthenticationToken(token)

            // 예외 발생 시 자동으로 AuthenticationEntryPoint로 감
            val authResult = authenticationProvider.authenticate(authToken)
            SecurityContextHolder.getContext().authentication = authResult
        }

        filterChain.doFilter(request, response)
    }
}